crates/topos-certificate-spammer/src/lib.rs

```diff
@@ -201,7 +201,18 @@ pub async fn run(
 
     // Is list of nodes is specified in the command line use them otherwise use
     // config file provided nodes
-    let target_nodes = if let Some(nodes) = args.target_nodes {
+    let target_nodes = if args.kubernetes {
+        // Kubernetes mode
+        if let (Some(dns), Some(number)) = (args.dns, args.number) {
+            (1..=number)
+                .map(|n| dns.replace("{N}", &n.to_string()))
+                .collect::<Vec<String>>()
```

Comment:
You can validate that the url is valid using [http::uri](https://docs.rs/http/1.0.0/http/uri/index.html) (and importing it in `Cargo.toml` `http.workspace = true` )

Reply:
Added a check. Debated how exactly we we want to check for a valid `Url`. So I think just doing it once in the beginning is the cheapest, cleanest way.